added support for multiple type package URIs in ConfigurableJsonResource

isalvadori opened the pull request:
- added support for multiple type package URIs in ConfigurableJsonResource

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

gg-dim committed:
Add sonatype snapshot url

maho7791 committed:
#3

- set context attribute of ROOT_ELEMENT is set

Signed-off-by: Mark Hoffmann <m.hoffmann@data-in-motion.biz>

maho7791 committed:
fixes #3

- added root_element attribute
- added tests

Signed-off-by: Mark Hoffmann <m.hoffmann@data-in-motion.biz>

maho7791 committed:
fixes #3

- added root_element attribute
- added tests

Signed-off-by: Mark Hoffmann <m.hoffmann@data-in-motion.biz>

isalvadori committed:
added support for multiple type package URIs in ConfigurableJsonResource

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

juergen-albert commented on org.gecko.emf.rest/src/org/gecko/emf/jaxrs/annotations/json/EMFJSONConfig.java:
```diff
@@ -87,9 +87,9 @@
-	 * return @see {@link EMFJs#OPTION_TYPE_PACKAGE_URI} default is eClass
+	 * return @see {@link EMFJs#OPTION_TYPE_PACKAGE_URIS} default is eClass
 	 */
-	String typePackageUri() default "";
```
This needs to be a String[], as we need to be able to hand over multiple URIs

isalvadori committed:
- updated basic model dependency

- updated tests to account for new basic model version

- typePackageUris is now a String[] with empty array as default

- update reader only if list of packages is not empty

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

isalvadori committed:
- removed gogo dependencies from bndrun

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

isalvadori committed:
- fix code smells

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

isalvadori committed:
- removed unnecessary cast

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

isalvadori committed:
added example model to test type package use options

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

isalvadori committed:
added tests for type package use

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

isalvadori committed:
look for package uris only in case of type use class and name

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

isalvadori committed:
fixed array case in getOrDefaultAsList and implemented tests

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

isalvadori committed:
added json test files to ignore list for license

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

isalvadori committed:
removed unused imports

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

isalvadori committed:
- added test cases for null package uri

-refactor tests

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

isalvadori committed:
- added test cases for OPTION_DATE_FORMAT

- added test cases for OPTION_INDENT_OUTPUT

- added test cases for OPTION_SERIALIZE_DEFAULT_VALUE

- added test cases for OPTION_SERIALIZE_TYPE/OPTION_TYPE_FIELD

- added test cases for OPTION_USE_ID

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

isalvadori committed:
refactor DateFormatTest to be time zone independent

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

isalvadori committed:
second try to make DateFormatTest independent from time zone

Signed-off-by: Ilenia Salvadori <i.salvadori@data-in-motion.biz>

juergen-albert commented:
@isalvadori I'm a bit lost here, as this is rather old. Is this done and can we merge it?

isalvadori commented:
@juergen-albert I am lost as well, as it has been quite a long time ago, but I guess, if I made the PR, it should be ready.
